### PR TITLE
React summernote rewrite

### LINKED ISSUE #4540: Rewrite react summernote
Currently, we are using react summernote for rich text field in React. There is a performance issue when there are multiple fields in a page (slow initial loading). 

Look into another alternative and implement it.

### PR DESCRIPTION
**Pull request by ekowidianto**
Opened on behalf of @syoopie 

Closes #4540 and #4596 

Replace react summernote with CKEditor

Old:
image

New:
image

**Improvements:**
1. Performance is better, not as laggy when loading many rich text editors at once
2. UI is more conformed to how coursemology looks
3. Image resizing is smoother
4. Removed Card from CommentCard, rich text editor tools no longer clip behind the card
5. Tables are now resizable

**Regressions:**
1. No more fullscreen & codeview
2. Codeblocks will not be rendered properly when editing existing rich text that was made with summernote. This is easily fixable by just using CKEditor to turn it into a code block again.
3. CKEditor will **NO LONGER SUPPORT MEDIA EMBEDS.** All existing media embeds will still display properly, but once they are edited they will disappear.

**Carried Over from summernote:**
1. Users are able to submit an empty comment (with spaces only, essentially the comment has nothing)
2. Users can edit a comment, make in blank, and then save it
3. Pressing and holding either a random key to type or backspace is slightly laggy as react tries to do an update for every single character change (https://css-tricks.com/debouncing-throttling-explained-examples/)
5. Error when using FormRichTextField: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
image

### CODE REVIEW

**Comment by ekowidianto** [comment]
@syoopie I have managed to make the scss file work. Could you help to solve the hound comments? Also see how I make the tests work - lmk if you have any question. I will review your code in a bit.

File: spec/support/capybara.rb [hunk 1]
```diff
@@ -43,26 +43,13 @@ def accept_confirm_dialog
...
-    # Special helper to fill in summernote textarea defined by react.
+    # Special helper to fill in draftjs textarea defined by react.
     #
     # Selector should specify the class of the target +textarea+, and method targets the +div+
     # within. This should change when the internals of the summernote react component is changed.
-    def fill_in_react_summernote(selector, text)
-      react_selector = ' + div.material-summernote > div > div[id^="react-summernote-"]'
-      fill_in_summernote(selector + react_selector, text)
+    def fill_in_draft_js(selector, text)
+      react_selector = ' + div.react-draft > div .public-DraftStyleDefault-block'
```
Comment (ekowidianto) [inline spec/support/capybara.rb:51]: Ref: https://spectrum.chat/testing-library/help/has-anyone-successfully-used-rtl-to-test-change-events-on-a-draftjs-text-box~f1928053-9e53-407a-9be5-70babeb4b692?m=MTU1NzUwMDM0NjAwMg==

**Review by ekowidianto — COMMENTED** [review]
Thanks! LGTM with minor nits
1) Remove all `// eslint-disable-next-line react/display-name` and add config to eslint
2) Remove draftjs from package
3) Remove extra features eg special characters

**Comment by ekowidianto** [comment]
@syoopie Could you document the incompatible html format regression and how you solve it as well?

File: client/app/bundles/course/leaderboard/components/tables/LeaderboardTable.tsx [hunk 2]
```diff
@@ -329,6 +329,7 @@ const LeaderboardTable: FC<Props> = (props: Props) => {
   );
 };
 
+// eslint-disable-next-line react/display-name
```
Comment (syoopie) [inline client/app/bundles/course/leaderboard/components/tables/LeaderboardTable.tsx:332]: This is still here cos even after globally disabling the error it still shows up for this particular file

File: client/app/bundles/course/assessment/skills/components/tables/SkillsTable.tsx [hunk 3]
```diff
@@ -349,6 +349,7 @@ const SkillsTable: FC<Props> = (props: Props) => {
   );
 };
 
+// eslint-disable-next-line react/display-name
```
Comment (syoopie) [inline client/app/bundles/course/assessment/skills/components/tables/SkillsTable.tsx:352]: Same for this

File: client/package.json [hunk 4]
```diff
@@ -55,6 +55,7 @@
     ],
     "moduleNameMapper": {
       ".scss$": "<rootDir>/SassStub.js",
+      "^.+\\.(css|less)$": "<rootDir>/CssStub.js",
```
Comment (ekowidianto) [inline client/package.json:58]: Do we need this?

File: client/app/lib/components/CKEditorRichText.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    name,
+    inputId,
+    clearOnSubmit,
+  } = props;
+
+  const [isFocused, setIsFocused] = useState(false);
+  const testFieldLabelColor = isFocused ? cyan[500] : undefined;
+
+  // Any type used as there is no documentation on the type
+  const [myEditor, setMyEditor] = useState<any>();
```
Comment (ekowidianto) [inline client/app/lib/components/CKEditorRichText.tsx:41]: Why not use `useState('')`?

File: client/app/lib/components/CKEditorRichText.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+  // Clear Editor field after comment is submitted, only if clearOnSubmit is true
+  useEffect(() => {
+    if (!disabled && clearOnSubmit && myEditor) {
+      myEditor.setData('');
```
Comment (ekowidianto) [inline client/app/lib/components/CKEditorRichText.tsx:46]: not `setMyEditor`?

**Review by ekowidianto — COMMENTED** [review]
Some nit comments

File: app/helpers/application_html_formatters_helper.rb [hunk 7]
```diff
@@ -121,6 +124,12 @@ def format_html(text)
     format_with_pipeline(DefaultHTMLPipeline, text)
   end
 
+  def format_ckeditor_rich_text(text)
+    DefaultHTMLPipeline.to_document("<div>#{text}</div>").child.inner_html.html_safe.
+      gsub(/<br>/, ''). # Remove <br> that causes table to be badly rendered
```
Comment (ekowidianto) [inline app/helpers/application_html_formatters_helper.rb:129]: To remove

**Pull request merged**